Framework/PythonInterface/test/python/plugins/algorithms/WorkflowAlgorithms/ReflectometryISISCreateTransmissionTest.py

```diff
@@ -0,0 +1,149 @@
+# Mantid Repository : https://github.com/mantidproject/mantid
+#
+# Copyright &copy; 2024 ISIS Rutherford Appleton Laboratory UKRI,
+#   NScD Oak Ridge National Laboratory, European Spallation Source,
+#   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+# SPDX - License - Identifier: GPL - 3.0 +
+import unittest
+import numpy as np
+
+from mantid import config
+from mantid.api import AnalysisDataService, MatrixWorkspace, WorkspaceGroup
+from mantid.simpleapi import CreateWorkspace
+from testhelpers import create_algorithm
+
+from plugins.algorithms.WorkflowAlgorithms.ReflectometryISISCreateTransmission import Prop
+
+
+class ReflectometryISISCreateTransmissionTest(unittest.TestCase):
+    _CONFIG_KEY_FACILITY = "default.facility"
+    _CONFIG_KEY_INST = "default.instrument"
+
+    _OUTPUT_WS_NAME = "out_ws"
+
+    _LOAD_ALG = "LoadAndMerge"
+    _FLOOD_ALG = "ApplyFloodWorkspace"
+    _BACK_SUB_ALG = "ReflectometryBackgroundSubtraction"
+    _TRANS_WS_ALG = "CreateTransmissionWorkspaceAuto"
+
+    def setUp(self):
+        self._oldFacility = config[self._CONFIG_KEY_FACILITY]
+        if self._oldFacility.strip() == "":
+            self._oldFacility = "TEST_LIVE"
+        self._oldInstrument = config[self._CONFIG_KEY_INST]
+        config[self._CONFIG_KEY_FACILITY] = "ISIS"
+        config[self._CONFIG_KEY_INST] = "POLREF"
+
+    def tearDown(self):
+        AnalysisDataService.clear()
+        config[self._CONFIG_KEY_FACILITY] = self._oldFacility
+        config[self._CONFIG_KEY_INST] = self._oldInstrument
+
+    def test_correct_output(self):
+        output_ws = self._run_algorithm(self._create_args("INTER13460", back_sub_roi=""))
+        self.assertIsNotNone(output_ws)
+        self.assertIsInstance(output_ws, MatrixWorkspace)
+        expected_history = [self._LOAD_ALG, self._FLOOD_ALG, self._TRANS_WS_ALG]
+        self._check_history(output_ws, expected_history)
+        self._check_output_data(output_ws, 771, 1.9796095404641e-07, 1.049193056445973e-05)
+
+    def test_correct_output_for_multiple_runs(self):
+        output_ws = self._run_algorithm(self._create_args(["INTER13463", "INTER13464"], back_sub_roi=""))
+        self.assertIsNotNone(output_ws)
+        self.assertIsInstance(output_ws, MatrixWorkspace)
+        expected_history = [self._LOAD_ALG, self._FLOOD_ALG, self._TRANS_WS_ALG]
+        self._check_history(output_ws, expected_history)
+        self._check_output_data(output_ws, 771, 0.018547099002195058, 1.2884763705979813e-05)
+
+    def test_correct_output_for_workspace_groups(self):
+        expected_history = [
+            self._LOAD_ALG,
+            self._FLOOD_ALG,
+            self._FLOOD_ALG,
+            self._BACK_SUB_ALG,
+            self._BACK_SUB_ALG,
+            self._TRANS_WS_ALG,
+            self._TRANS_WS_ALG,
+        ]
```

Comment:
I think it is possible to refactor the history recreation list in a more compact form (similar in other lists of the file):
```suggestion
        expected_history = [self._LOAD_ALG, *[self._FLOOD_ALG]*2,                                          *[self._BACK_SUB_ALG]*2, *[self._TRANS_WS_ALG]*2]

```
but since the list is not that long, and it is clear enough as it is. I leave this change up to you.

Reply:
That's interesting syntax that I don't think I've noticed before, thanks for mentioning. It looks like you'd be creating a series of lists this way that all end up in one? I think, as you say, because this list is not that long I might leave it as it is because I think it's currently quite straightforward to see what's going on. Handy to know about this for longer lists, though, as I can definitely see there would be an advantage there.